hazelcast/src/main/java/com/hazelcast/datalink/impl/hazelcastdatalink/HazelcastDataLinkConfigLoader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink.impl.hazelcastdatalink;
+
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.hazelcast.datalink.HazelcastDataLink.CLIENT_XML;
+import static com.hazelcast.datalink.HazelcastDataLink.CLIENT_XML_PATH;
+import static com.hazelcast.datalink.HazelcastDataLink.CLIENT_YML_PATH;
+import static com.hazelcast.datalink.HazelcastDataLink.CLIENT_YML;
+
+/**
+ * Loads the DataLinkConfig if file path is provided
+ */
+public class HazelcastDataLinkConfigLoader {
+
+    private static final ILogger LOGGER = Logger.getLogger(HazelcastDataLinkConfigLoader.class);
+
+    public DataLinkConfig loadConfigFromFile(DataLinkConfig dataLinkConfig) {
+        // Make a copy to preserve the original configuration
+        DataLinkConfig loadedDataLinkConfig = new DataLinkConfig(dataLinkConfig);
+
+        String clientXmlPath = dataLinkConfig.getProperty(CLIENT_XML_PATH);
+        if (loadConfig(loadedDataLinkConfig, clientXmlPath, CLIENT_XML)) {
```

Comment:
you read property for paths and just push key for strings. I think you should go one way, I think it could be:
`if (loadConfig(loadedDataLinkConfig, CLIENT_XML_PATH, CLIENT_XML)) {`

Reply:
rev : e5a7a7c91d81a0833662b44490af811399248446